src/50_Samples_%26_Templates/Product.html

```diff
@@ -313,11 +415,20 @@ <h5 class="price-description">Price: $1.99</h5>
         </div>
         <div>
           <label for="size">Size:</label>
-          <amp-selector name="size" layout="container">
+          <amp-selector name="size" layout="container" on="select:AMP.setState(selectedSize=event.targetOption)" [selected]="allColors[selectedCarousel].availableSizes.indexOf(selectedSize) >= 0 ? selectedSize : allColors[selectedCarousel].availableSizes[0]">
             <ul>
-              <li option="S">S</li>
-              <li option="M" selected>M</li>
+              <div class="size-container">
+                <li option="S">S</li>
+                <span class="show-size-unavailable" [class]="allColors[selectedCarousel].availableSizes.indexOf('S') >= 0 ? 'hide' : 'show-size-unavailable'">X</span>
```

Comment:
I usually try to avoid using html/text (which should be for content/structure) for styling and use css instead. The problem is, for example, that the X doesn't fit correctly into the square (e.g. the lines don't go into the corners).